### ISSUE TITLE
getStatus request unresponsive when booting up an ios simulator

### ISSUE THREAD

**Issue by StoneTree1** [appium/appium#6548]
## The problem

Appium does not respond to a status request (http://appiumserver:4723/wd/hub/status) while it is starting up a new simulator for a test. This is a problem when using SeleniumGridExtras as that has code to take down nodes when they become unresponsive. Because the status request doesn't reply the grid thinks the node has gone offline and it takes it down just as the test is initialing.
## Environment
- Appium version (or git revision) that exhibits the issue:Appium 1.5.2
- Last Appium version that did not exhibit the issue (if applicable):Appium 1.4.16
- Desktop OS/version used to run Appium:OSX 10.11
- Node.js version (unless using Appium.app|exe):5.11.0
- Mobile platform/version under test:IOS 9.1
- Real device or emulator/simulator: Simulator
- Appium CLI or Appium.app|exe: CLI
## Link to Appium logs

You can see on line 81-83 the Status request is closed by the grid after 10 seconds without a response
https://gist.github.com/StoneTree1/7c02b9685ad33a78751babdedf752ed9
## Code To Reproduce Issue [ Good To Have ]
- run a test against appium using an ios simulator
- while it is loading up the simulator perform a status request (http://yourappiumserver.com:4723/wd/hub/status). Easiest if you use a browser for this.
- You will see the request will not complete until the simulator is finished loading.

**Comment by dpgraham** [comment]
Hi @StoneTree1, I can't reproduce this issue. Is this still a problem?

**Comment by StoneTree1** [comment]
I haven't seen this problem with appium 1.5.3 so I think this can be closed

**Comment by StoneTree1** [comment]
I'm seeing this issue again in Appium version 1.6.4-Beta. This was not limited to ios simulator. I saw with Android also.

The response does come once app install is complete but this can be after a considerable time and may impact grid node polling. See this thread on appium discuss https://discuss.appium.io/t/selenium-grid-3-0-1-marking-the-node-as-down/14760

**Comment by mykola-mokhnach** [comment]
@StoneTree1 Can you please try to apply [this patch](https://github.com/appium/appium-xcuitest-driver/pull/364) to your local source and check how Grid behaves?

**Comment by StoneTree1** [comment]
After editing the files do I need to do a build or can I just run Appium directly?

**Comment by mykola-mokhnach** [comment]
yes you need to build it. 

Execute

```
npm install gulp
gulp once
```

to build in the root folder of the module. gulp will raise some errors first and will ask you to install some additional required modules. Your target is to have the **transpile** stage completed.

Then just kill the appium/node process to make sure your old code is not cached and you are ready to go.

**Comment by StoneTree1** [comment]
I gave that a try. It appeared to transpile correctly. But I still see a delay getting the Status response. See the log here
https://gist.github.com/StoneTree1/d22ad07aabd99ca0f6c50d9a89087849

The GET /wd/hub/status at line 38 gets a response at line 132

**Comment by mykola-mokhnach** [comment]
This PR has been recently merged to beta. Can you please try it with the recent beta build?

```
npm remove -g appium
npm install -g appium@1.6.4-beta
```

**Comment by StoneTree1** [comment]
I have tried with latest beta but I still see the delayed response to the status call. The installed Appium version
Welcome to Appium v1.6.4-beta (REV 05efd70dc350bf60c47d25d1d48da188ffb504df)

**Comment by mykola-mokhnach** [comment]
ok, I did some additional fixes, but they are still waiting for a review. I'll let you know when these are merged. 
you'll just have to reinstall beta to get the most recent changed.

**Comment by mykola-mokhnach** [comment]
Ok, the change is in master. Please reinstall 1.6.4-beta and try it.

**Comment by StoneTree1** [comment]
I reinstalled Appium but I'm not seeing your changes in the files pulled by npm.
https://github.com/appium/appium-xcuitest-driver/pull/366/files

**Comment by mykola-mokhnach** [comment]
:( probably because it has no version tag set yet. Wait a bit until version is set or you can try to apply the changes manually - you know how.

**Comment by mykola-mokhnach** [comment]
The change is published in 1.6.4-beta and is ready for use.
